csrc/kernel_cache.cpp

```diff
@@ -997,6 +1021,11 @@ FusionKernelRuntime::FusionKernelRuntime(
   // Pre-compute the executor order so that the run time path
   //  would go directly to kernel launch.
   prepareRuntimeOrder();
+
+  // Restore State from Flatbuffers
+  if (serde_buffer != nullptr) {
+    deserialize(serde_buffer);
+  }
```

Comment:
I'm surprised this block is not at line 1008. Is there a good reason to put it here? Logically, it makes more sense to 

```
if (buffer is null) {
  do the expensive work
} else {
  restore from the buffer
}
```

Reply:
Gotcha! This is `FusionKernelRuntime::deserialize` not `SegmentedFusion::deserialize`. 

A follow-up question: as you said, `FusionKernelRuntime::deserialize` creates `FusionExecutor`s from the flatbuffer. So aren't we supposed to have an if-else branch that creates `FusionExecutor`s in a different way? Conceptually, it makes more sense to 

```
if (serialization is on) {
  create something from flatbuffer;
} else {
  create something from void;
}
```

But I don't see the if-else branch in your code.

Reply:
The `else` branch occurs at https://github.com/NVIDIA/Fuser/blob/main/csrc/kernel_cache.cpp#L476-L478. There isn't an obvious reason why it couldn't occur in the constructor.

In any case, I moved the if-branch back to its original position at
https://github.com/NVIDIA/Fuser/blob/main/csrc/kernel_cache.cpp#L924-L926.